File: 1주차/김태경/카펫.java
//가로를 x, 세로를 y라고 했을때 아래 두 방정식을 최적화한다.
// brown = a, yellow = b
// (x-2)(y-2)=a
// xy = a+b
// -> 2x+2y = a;

import java.util.*;
class Solution {
    public int[] solution(int brown, int yellow) {
        int[] answer = {};
        int totalSum = brown+yellow;
        ArrayList<int[]> array = new ArrayList<>();
        chck(totalSum, array);
        answer = find(brown, array);
        return answer;
    }
    
    //brown+yellow의 약수를 찾아 저장
    public void chck(int totalSum, ArrayList<int[]> array){
        for(int i=1;i<=Math.sqrt(totalSum);i++){
            if(totalSum%i==0){//약수
                array.add(new int[]{totalSum/i,i});
            }
        }
    }
    
    //값 후보인 pair 중 식을 만족하는 값 찾기
    public int[] find(int brown,ArrayList<int[]> array){
        for(int[] pair : array){
            //2x+2y-4 = brown 식을 만족하는 pair찾기
            int value = 2*pair[0] + 2*pair[1] - 4;
            if(value==brown){
                return pair;
            }
        }
        return new int[]{0,0};
        
        
    }
    
}

File: 1주차/박수교/카펫.py
import math
def solution(brown, yellow):
    if math.sqrt(yellow) == int(math.sqrt(yellow)):
        return [math.sqrt(yellow)+2, math.sqrt(yellow)+2]
    
    #가로+세로 값
    n = (brown - 4) // 2
    
    for i in range(1,n+1):
        if (i * (n-i)) == yellow:
            return [i+2, (n-i)+2] if i > n-i else [(n-i)+2, i+2]
